pytorch_to_returnn/torch/nn/functional.py

```diff
@@ -46,6 +47,14 @@ def cast(input: Union[_T, Tensor, _number], dtype: Union[str, _dtype]) -> Union[
     return input
   return modules.Cast(dtype=dtype)(input)
 
+def cat(tensors, dim=0):
+  from .modules.operator import Copy
+  naming = Naming.get_instance()
+  for tensor in tensors:
+    returnn_data = naming.tensors[tensor].returnn_data
+    assert returnn_data.get_axis_from_description(dim) == returnn_data.feature_dim_axis, "Concatenation in dimensions other than the feature dimension is currently not supported."
```

Comment:
This is wrong, in multiple ways.

`get_axis_from_description` doesn't make any sense here. `dim` is a Torch dimension (integer). `get_axis_from_description` is for RETURNN axes descriptions (usually string).

Anyway, move this logic to the `Cat` module (see my other comment). And in there, handle the axis mapping (Torch -> RETURNN) as we do in the other layers. And then you can check that the RETURNN axis (int) matches the feature dim axis for all the given tensors, and if not, fail with some exception.